src/pages/signin/SignInPage.js

```diff
@@ -91,21 +123,39 @@ function SignInPage({credentials, account, isInModal}) {
      * and we need it here since welcome text(`welcomeText`) also depends on it */
     const [isUsingRecoveryCode, setIsUsingRecoveryCode] = useState(false);
 
+    /** This state is needed to keep track of whether the user has opted to use magic codes
+     * instead of signing in via SAML when SAML is enabled and not required */
+    const [isUsingMagicCode, setIsUsingMagicCode] = useState(false);
+
     useEffect(() => Performance.measureTTI(), []);
-    useEffect(() => {
-        App.setLocale(Localize.getDevicePreferredLocale());
-    }, []);
-
-    const {shouldShowLoginForm, shouldShowEmailDeliveryFailurePage, shouldShowUnlinkLoginForm, shouldShowValidateCodeForm, shouldShowWelcomeHeader, shouldShowWelcomeText} = getRenderOptions(
-        {
-            hasLogin: Boolean(credentials.login),
-            hasValidateCode: Boolean(credentials.validateCode),
-            hasAccount: !_.isEmpty(account),
-            isPrimaryLogin: !account.primaryLogin || account.primaryLogin === credentials.login,
-            isAccountValidated: Boolean(account.validated),
-            hasEmailDeliveryFailure: Boolean(account.hasEmailDeliveryFailure),
-        },
-    );
+    useEffect(() => App.setLocale(Localize.getDevicePreferredLocale()), []);
+
+    const {
+        shouldShowLoginForm,
+        shouldShowEmailDeliveryFailurePage,
+        shouldShowUnlinkLoginForm,
+        shouldShowValidateCodeForm,
+        shouldShowChooseSSOOrMagicCode,
+        shouldInitiateSAMLLogin,
+        shouldShowWelcomeHeader,
+        shouldShowWelcomeText,
+    } = getRenderOptions({
+        hasLogin: Boolean(credentials.login),
+        hasValidateCode: Boolean(credentials.validateCode),
+        hasAccount: !_.isEmpty(account),
+        isPrimaryLogin: !account.primaryLogin || account.primaryLogin === credentials.login,
+        isAccountValidated: Boolean(account.validated),
+        isSAMLEnabled: Boolean(account.isSAMLEnabled),
+        isSAMLRequired: Boolean(account.isSAMLRequired),
```

Comment:
@situchan manually setting these to true (or false, as necessary) will simulate the user having SAML set up on a private domain. If you are having difficulty getting that set up for testing, you can just edit these to go through the flow - though of course then the actual SSO login testing + callback will not work 😅